store/screenSize/reducer.js

```diff
@@ -0,0 +1,25 @@
+import breakpoints from 'common/styles/breakpoints';
+import initialState from '../initialState';
+import { SCREEN_RESIZE } from './types';
+
+const reducer = (state = initialState.screenSize, action) => {
+  const { height, width } = action;
+
+  switch (action.type) {
+    case SCREEN_RESIZE:
+      return {
+        ...state,
+        width,
+        height,
+        isXs: width >= (breakpoints.xs || 0) && width < breakpoints.sm,
+        isSm: width >= breakpoints.sm && width < breakpoints.md,
+        isMd: width >= breakpoints.md && width < breakpoints.lg,
+        isLg: width >= breakpoints.lg && width < breakpoints.xl,
+        isXl: width >= breakpoints.xl,
+      };
```

Comment:
Ah. I'm not a fan of Immutable.js personally. I'd prefer just using ES6 paradigms like the spread operator to ensure immutability. I liked that other library you brought up since imports are easier than decorating a component with a Redux connection for newer devs, but since we'll be using Redux to manage registration form connection to localStorage, might as well just keep it going this way.